scipy/stats/_stats_py.py

```diff
@@ -7734,29 +7714,29 @@ def _attempt_exact_2kssamp(n1, n2, g, d, alternative):
         return True, d, 1.0
     saw_fp_error, prob = False, np.nan
     try:
-        if alternative == 'two-sided':
-            if n1 == n2:
-                prob = _compute_prob_outside_square(n1, h)
-            else:
-                prob = _compute_outer_prob_inside_method(n1, n2, g, h)
-        else:
-            if n1 == n2:
-                # prob = binom(2n, n-h) / binom(2n, n)
-                # Evaluating in that form incurs roundoff errors
-                # from special.binom. Instead calculate directly
-                jrange = np.arange(h)
-                prob = np.prod((n1 - jrange) / (n1 + jrange + 1.0))
+        with np.errstate(invalid="raise", over="raise"):
+            if alternative == 'two-sided':
+                if n1 == n2:
+                    prob = _compute_prob_outside_square(n1, h)
+                else:
+                    prob = _compute_outer_prob_inside_method(n1, n2, g, h)
             else:
-                with np.errstate(over='raise'):
-                    num_paths = _count_paths_outside_method(n1, n2, g, h)
-                bin = special.binom(n1 + n2, n1)
-                if not np.isfinite(bin) or not np.isfinite(num_paths)\
-                        or num_paths > bin:
-                    saw_fp_error = True
+                if n1 == n2:
+                    # prob = binom(2n, n-h) / binom(2n, n)
+                    # Evaluating in that form incurs roundoff errors
+                    # from special.binom. Instead calculate directly
+                    jrange = np.arange(h)
+                    prob = np.prod((n1 - jrange) / (n1 + jrange + 1.0))
                 else:
-                    prob = num_paths / bin
-
-    except FloatingPointError:
+                    with np.errstate(over='raise'):
+                        num_paths = _count_paths_outside_method(n1, n2, g, h)
+                    bin = special.binom(n1 + n2, n1)
+                    if num_paths > bin:
```

Comment:
```suggestion
                    if num_paths > bin or np.isinf(bin):
```
In the failing test, `num_paths` is finite but `bin` is not, and there is no overflow warning. 
``python3
from scipy import special
special.binom(369 + 881, 369)  # inf, no warning
```
In the (very unlikely) case that `num_paths` is only slightly smaller than the true value of `binom(n1 + n2, n1)`, this will incorrectly report that the p-value is zero. I think it would be safer to make `saw_fp_error = True` and revert to the normal approximation as the code would do now. 

Alternatively, we could consider computing with logs, e.g. `logbinom(n, k) = loggamma(n+1) - loggamma(n-k+1) - loggamma(k+1)`, although that should be a separate enhancement.

Reply:
@Balthasar-eu I went ahead and pushed this change, but do you agree?